### ISSUE TITLE
Compile error when compiling wonka 

### ISSUE THREAD

**Issue by jfrolich** [melange-re/melange#320]
When compiling our repo we get a puzzling compile error of one of the dependencies: Wonka. This is an observable library that is being used by URQL written in ReasonML.

The compile error is as follows:

```
~/d/wonka ~> esy b                       0.251s (melange-compile-error|✚?) 15:05
info esy build 0.6.12 (using esy.json)
File "dune.mel", line 72, characters 1-282:
72 |  (rule
73 | (targets  Wonka_operators.ast)
74 |  (deps (:inputs  Wonka_operators.re)  ../.melange.eobjs/.bsbuild ../.melange.eobjs/.sourcedirs.json)
75 | (action
76 |  (run melc  -w a -bs-super-errors -bs-no-version-header -absname -bs-ast -o %{targets} %{inputs}))
77 | (enabled_if %{bin-available:melc}) )
Error: No rule found for .melange.eobjs/.sourcedirs.json
Waiting for filesystem changes...   
``` 

I managed to reproduce it in the original repo: https://github.com/jfrolich/wonka/tree/melange-compile-error

**Comment by anmonteiro** [comment]
This should be fixed after #343, but it needs a 4.14 compatible gentype

**Comment by anmonteiro** [comment]
This one is fixed, let's track gentype support in https://github.com/melange-re/melange/issues/84
